addons/rmsmartshape/shapes/index_map.gd

```diff
@@ -239,11 +239,13 @@ static func indicies_to_edges(p_indicies: Array[int]) -> Array:
 	return edges
 
 
+## Returns a Dict[Variant, Array[SS2D_IndexMap]]
 static func index_map_array_sort_by_object(imaps: Array) -> Dictionary:
 	var dict := {}
-	for imap in imaps:
-		if not dict.has(imap.object):
-			var arr: Array[SS2D_IndexMap] = []
+	for imap: SS2D_IndexMap in imaps:
+		var arr: Array[SS2D_IndexMap] = dict.get(imap.object)
```

Comment:
Can't assign null to an Array, so it needs to be checked if the dictionary contains such an object.